app/src/main/java/com/cmput301f20t21/bookfriends/repositories/BookRepository.java

```diff
@@ -93,6 +93,17 @@ public Book getBookFromDocument(DocumentSnapshot document) {
         return bookCollection.whereEqualTo("owner", username).get();
     }
 
+//  TODO ============ the possibly ideal way of us mapping doc directly to entities, works now but not enabled in this PR ========
+//    public Task<ArrayList<Book>> getBooksOfOwnerIdParsed(String username) {
+//        return bookCollection.whereEqualTo("owner", username).get().continueWith(qSnap -> {
+//            List<DocumentSnapshot> docs = qSnap.getResult().getDocuments();
+//            return (ArrayList<Book>) docs
+//                    .stream()
+//                    .map(doc -> doc.toObject(Book.class))
+//                    .collect(Collectors.toList());
+//        });
+//    }
+
```

Comment:
we can potentially make repo APIs all like this (mapping directly to entity data)